web-components/src/components/audio-player/AudioPlayer.ts

```diff
@@ -0,0 +1,354 @@
+/**
+ * Copyright (c) Cisco Systems, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE src in the root directory of this source tree.
+ *
+ */
+
+import { customElementWithCheck } from "@/mixins/CustomElementCheck";
+import { html, internalProperty, LitElement, property, PropertyValues } from "lit-element";
+import { classMap } from "lit-html/directives/class-map";
+import styles from "./scss/module.scss";
+import reset from "@/wc_scss/reset.scss";
+import { nothing } from "lit-html";
+
+export type AudioEvent = {
+  name: string;
+  onEvent: EventListener;
+};
+
+export const playbackSpeeds = [
+  {
+    value: 0.25,
+    label: "0.25"
+  },
+  {
+    value: 0.5,
+    label: "0.50"
+  },
+  {
+    value: 0.75,
+    label: "0.75"
+  },
+  {
+    value: 1,
+    label: "Normal"
+  },
+  {
+    value: 1.25,
+    label: "1.25"
+  },
+  {
+    value: 1.5,
+    label: "1.50"
+  },
+  {
+    value: 1.75,
+    label: "1.75"
+  },
+  {
+    value: 2.0,
+    label: "2"
+  }
+];
+
+export type Label = {
+  ariaLabel: string;
+  tooltipText: string;
+};
+
+export type LabelMap = {
+  playBtn: Label;
+  duration: Label;
+  timeline: Label;
+  volumeBtn: Label;
+  volumeSlider: Label;
+  playbackSpeedBtn: Label;
+};
+
+export const defaultLabelMap = {
+  playBtn: {
+    ariaLabel: "Play / Pause",
+    tooltipText: "Play / Pause"
+  },
+  duration: {
+    ariaLabel: "Duration:",
+    tooltipText: "Duration"
+  },
+  timeline: {
+    ariaLabel: "Audio Seekbar",
+    tooltipText: "Audio Seekbar"
+  },
+  volumeBtn: {
+    ariaLabel: "Mute / Unmute Audio",
+    tooltipText: "Mute / Unmute"
+  },
+  volumeSlider: {
+    ariaLabel: "Volume level",
+    tooltipText: "Volume level"
+  },
+  playbackSpeedBtn: {
+    ariaLabel: "Playback speed",
+    tooltipText: "Playback speed"
+  }
+};
+
+export namespace AudioPlayer {
+  @customElementWithCheck("md-audio-player")
+  export class ELEMENT extends LitElement {
+    @property({ type: String }) src = "";
+    @property({ type: Object }) labelMap: LabelMap;
+    @internalProperty() events: AudioEvent[] = [];
+    @internalProperty() audio!: HTMLAudioElement;
```

Comment:
you can remove the non null assertion ```!``` since you are now defining this in the constructor